test/close.test.js

```diff
@@ -138,7 +138,39 @@ test('onClose should keep the context', t => {
   })
 })
 
-test('Should return error while closing - injection', t => {
+test('Should return error while closing (promise) - injection', t => {
+  t.plan(4)
+  const fastify = Fastify()
+
+  fastify.addHook('onClose', (instance, done) => {
+    setTimeout(done, 150)
```

Comment:
is this needed? Can you use a `setImmediate` instead? otherwise use fake-timers as timeouts make tests flaky

Reply:
I tried with:

```diff
+ fastify.addHook('onClose', (instance, done) => { done() })
...
t.strictEqual(res.statusCode, 200)
fastify.close()
+ process.nextTick(() => {
+     fastify.inject({
+       method: 'GET',
+        url: '/'
+      }).catch(err => {
+        t.ok(err)
+        t.equal(err.message, 'Server is closed')
+      })
+ })
```
and works well, but I don't know if is it the best choice. Maybe can help you.

Reply:
Hmm, to be honest, I just keep it to have consistency across the tests as the test is quite similar as the one before but being honest, I think is not needed as we're trying test error after closing through `fastify.inject`, it doesn't have sense add a timer at all and @RafaelGSS approach looks good and a way to go, what do you think?

Reply:
+1